src/DGtal/io/colormaps/TickedColorMap.ih

```diff
@@ -0,0 +1,211 @@
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+
+/**
+ * @file TickedColorMap.ih
+ * @author David Coeurjolly (\c david.coeurjolly@liris.cnrs.fr )
+ * Laboratoire d'InfoRmatique en Image et Systemes d'information - LIRIS (CNRS, UMR 5205), CNRS, France
+ *
+ * @date 2015/04/06
+ *
+ * Implementation of inline methods defined in TickedColorMap.h
+ *
+ * This file is part of the DGtal library.
+ */
+
+
+//////////////////////////////////////////////////////////////////////////////
+#include <cstdlib>
+#include <boost/bind.hpp>
+//////////////////////////////////////////////////////////////////////////////
+
+///////////////////////////////////////////////////////////////////////////////
+// IMPLEMENTATION of inline methods.
+///////////////////////////////////////////////////////////////////////////////
+
+///////////////////////////////////////////////////////////////////////////////
+// ----------------------- Standard services ------------------------------
+
+
+template <typename Value, typename CMAP>
+inline
+DGtal::TickedColorMap<Value,CMAP>::TickedColorMap
+( const Value & amin,
+  const Value & amax,
+  const Color & color )
+  : myMin( amin ), myMax( amax ), myTickColor( color )
+{
+  ASSERT(myMin < myMax);
+  myColorMap = new ColorMap(myMin,myMax);
+}
+
+
+template <typename Value, typename CMAP>
+inline
+DGtal::TickedColorMap<Value,CMAP>::TickedColorMap
+( const TickedColorMap<Value,CMAP> & other )
+  : myMin( other.myMin ), myMax( other.myMax ), myTickColor( other.myTickColor )
+{ 
+}
+
+template <typename Value, typename CMAP>
+inline
+DGtal::TickedColorMap<Value,CMAP>::~TickedColorMap()
+{
+  delete myColorMap;
+}
+
+template <typename Value, typename CMAP>
+DGtal::TickedColorMap<Value,CMAP> &
+DGtal::TickedColorMap<Value,CMAP>::operator=
+( const TickedColorMap<Value,CMAP> & other )
+{
+  if ( &other != this ) {
+    myMin = other.myMin;
+    myMax = other.myMax;
+    myTickColor = other.myTickColor;
+    myTicks = other.myTicks;   
+  }
+  return *this;
+}
+
+///////////////////////////////////////////////////////////////////////////////
+// Interface - public :
+
+template<typename Value, typename CMAP>
+inline
+const Value &
+DGtal::TickedColorMap<Value,CMAP>::min() const
+{
+  return myMin;
+}
+
+template<typename Value, typename CMAP>
+inline
+const Value &
+DGtal::TickedColorMap<Value,CMAP>::max() const
+{
+  return myMax;
+}
+
+template<typename Value, typename CMAP>
+inline
+DGtal::Color
+DGtal::TickedColorMap<Value,CMAP>::operator()( const Value & value ) const
+{
+  typename std::vector< std::pair<Value,Value> >::const_iterator  up;
+  
+  //Test the upper bound
+  up = std::upper_bound(myTicks.begin(), myTicks.end(), std::make_pair(value,value));
+  if (up == myTicks.end())
+  {
+    if (std::abs(value - myTicks[myTicks.size() -1].first) < myTicks[myTicks.size() -1].second)
```

Comment:
this line segfault if one try to use the ticked colormap without inserting ticks.
IMO, this should at least be asserted.
adding ASSERT( myTicks.size() ); at the beginning of operator() for example.